File: InvertBinaryTree/invert.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return root
	}

	leftNode, rightNode := root.Left, root.Right

	root.Left = rightNode
	root.Right = leftNode

	invertTree(root.Left)
	invertTree(root.Right)

	return root
}

File: InvertBinaryTree/invert.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */

const invertTree = root => {
  // break case
  if (root === null) return root;

  let left = root.left;
  root.left = root.right;
  root.right = left;

  invertTree(root.left);
  invertTree(root.right);

  return root;
};
